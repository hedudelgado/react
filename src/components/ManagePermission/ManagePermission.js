
import React from 'react';
import { object } from 'prop-types';
import Button from '../../layout/Button';
import Typography from '../../layout/Typography';
import { managePermissionLink } from '../../config.json';
import { getJsonItem } from '../../utils/session-storage';

const ManagePermission = ({
  dictionary,
}) => {
  const redirectToCustomerPreference = () => {
    window.location.assign(`${getJsonItem('environment').domain}${managePermissionLink}`);
  };

  return (
    <section className="component-container">

      <Typography component="h4" className="pi-color-greyscale-darkest">
        {dictionary['managepermission.title']}
      </Typography>

      <Typography mb="2">
        {dictionary['managepermission.description']}
      </Typography>

      <Button
        data-test="manage-permission"
        color="secondary"
        fullWidth
        onClick={redirectToCustomerPreference}
      >
        {dictionary['managepermission.button']}
      </Button>

    </section>
  );
};

ManagePermission.propTypes = {
  dictionary: object.isRequired,
};

export default ManagePermission;
