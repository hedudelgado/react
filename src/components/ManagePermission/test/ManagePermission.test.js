
import React from 'react';
import { shallow } from 'enzyme';
import ManagePermission from '../ManagePermission';

jest.mock('../../../utils/session-storage', () => ({
  __esModule: true,
  getJsonItem: () => ({
    domain: 'https://www.premierinn.com.de-qa.nativ-systems.com',
  }),
}));

describe('ManagePermission Component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      dictionary: {
        'managepermission.title': 'Contact and permissions centre',
        'managepermission.button': 'Manage permissions',
      },
      redirectToCustomerPreference: jest.fn(),
    };
    wrapper = shallow(<ManagePermission {...props} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show the Manage Permission Button', () => {
    expect(wrapper).toContainExactlyOneMatchingElement('[data-test="manage-permission"]');
  });

  it('should trigger the link when clicking the Manage Permission button', () => {
    window.location.assign = jest.fn();
    wrapper.find('[data-test="manage-permission"]').simulate('click');
    expect(window.location.assign).toHaveBeenCalledWith('https://www.premierinn.com.de-qa.nativ-systems.com/gb/en/home.html');
  });
});
