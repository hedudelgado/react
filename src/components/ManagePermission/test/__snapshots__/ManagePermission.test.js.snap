// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManagePermission Component should match snapshot 1`] = `
<section
  className="component-container"
>
  <Typography
    className="pi-color-greyscale-darkest"
    component="h4"
  >
    Contact and permissions centre
  </Typography>
  <Typography
    component="p"
    mb="2"
  />
  <Button
    color="secondary"
    data-test="manage-permission"
    fullWidth={true}
    href={null}
    onClick={[Function]}
    submit={false}
    variant="button"
  >
    Manage permissions
  </Button>
</section>
`;
